monai/bundle/scripts.py

```diff
@@ -172,3 +224,88 @@ def verify_metadata(
+    config_file: Optional[Union[str, Sequence[str]]] = None,
+    device: Optional[str] = None,
+    p: Optional[int] = None,
+    n: Optional[int] = None,
+    any: Optional[int] = None,
+    args_file: Optional[str] = None,
+    **override,
+):
+    """
+    Verify the input and output data shape and data type of network defined in the metadata.
+    Will test with fake Tensor data according to the network args with id:
+    input data:
+    "_meta_#network_data_format#inputs#image#num_channels"
+    "_meta_#network_data_format#inputs#image#spatial_shape"
+    "_meta_#network_data_format#inputs#image#dtype"
+    output data:
+    "_meta_#network_data_format#outputs#pred#num_channels"
+    "_meta_#network_data_format#outputs#pred#dtype"
+
+    Args:
+        net_id: ID name of the network component to verify, it must be `torch.nn.Module`.
+        meta_file: filepath of the metadata file to get network args, if `None`, must be provided in `args_file`.
+            if it is a list of file paths, the content of them will be merged.
+        config_file: filepath of the config file to get network definition, if `None`, must be provided in `args_file`.
+            if it is a list of file paths, the content of them will be merged.
+        device: target device to run the network forward computation, if None, prefer to "cuda" if existing.
+        p: power factor to generate fake data shape if dim of expected shape is "x**p", default to 1.
+        p: multiply factor to generate fake data shape if dim of expected shape is "x*n", default to 1.
+        any: specified size to generate fake data shape if dim of expected shape is "*", default to 1.
+        args_file: a JSON or YAML file to provide default values for `meta_file`, `config_file`,
+            `net_id` and override pairs. so that the command line inputs can be simplified.
+        override: id-value pairs to override or add the corresponding config content.
+            e.g. ``--_meta#network_data_format#inputs#image#num_channels 3``.
+
+    """
+
+    _args = _update_args(
+        args=args_file,
+        net_id=net_id,
+        meta_file=meta_file,
+        config_file=config_file,
+        device=device,
+        p=p,
+        n=n,
+        any=any,
+        **override,
+    )
+    _log_input_summary(tag="verify_net_in_out", args=_args)
+
+    parser = ConfigParser()
+    parser.read_config(f=_args.pop("config_file"))
+    parser.read_meta(f=_args.pop("meta_file"))
+    id = _args.pop("net_id", "")
+    device_ = torch.device(_args.pop("device", "cuda" if torch.cuda.is_available() else "cpu"))
+    p = _args.pop("p", 1)
+    n = _args.pop("n", 1)
+    any = _args.pop("any", 1)
+
+    # the rest key-values in the _args are to override config content
+    for k, v in _args.items():
+        parser[k] = v
+
+    net = parser.get_parsed_content(id).to(device_)
+    input_channels = parser["_meta_#network_data_format#inputs#image#num_channels"]
```

Comment:
If one of the field is not provided, it could be a warning instead of key error?

Reply:
Thanks for your suggestion.
I think we should raise an exception to stop the verification if missing the keys, I enhanced the error log to make it more clear in the latest commit.

Thanks.